x-pack/plugins/actions/server/plugin.ts

```diff
@@ -442,6 +440,8 @@ export class ActionsPlugin implements Plugin<Promise<PluginSetupContract>, Plugi
 
     return async function actionsRouteHandlerContext(context, request) {
       const [{ savedObjects }, { taskManager }] = await core.getStartServices();
+      const defaultKibanaIndex = (await config.pipe(first()).toPromise()).kibana.index;
```

Comment:
I think this will suffer from a couple of problems that I'd suggest double checking. 🤔 

A call to `.pipe(first()).toPromise()` doesn't get _the first ever emitted event_, but rather _the first emitted event from **now**_.
This means that each call to `actionsRouteHandlerContext` will actually block until the config updates... I have no idea how often Core emits a new config, but I don't think we should be waiting on it.

Additionally, this will cause a whole Observable -> Promise chain on every call to `actionsRouteHandlerContext`.

It might be better to move this to line 440, so it's outside of `actionsRouteHandlerContext` and only has to happen once. You can then reference that promise and `await` it on each call.